src/faebryk/core/node.py

```diff
@@ -709,3 +743,48 @@ def zip_children_by_name_with[N: Node](
     @staticmethod
     def with_names[N: Node](nodes: Iterable[N]) -> dict[str, N]:
         return {n.get_name(): n for n in nodes}
+
+
+def magic_pointer[O: Node](out_type: type[O] | None = None) -> O:
```

Comment:
Does this belong in `Node`? It'd be the only `classmethod` a dev would typically touch in it, but it's also fewer imports and precisely node-related, so maybe 🤷‍♂️